isis/tests/FunctionalTestsChan1m32isis.cpp

```diff
@@ -0,0 +1,550 @@
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  ASSERT_EQ(inst["SpacecraftName"][0].toStdString(), "CHANDRAYAAN-1");
+  ASSERT_EQ(inst["InstrumentId"][0].toStdString(), "M3" );
+  ASSERT_EQ(inst["TargetName"][0].toStdString(), "MOON" );
+  ASSERT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "12/1759056.764" );
+  ASSERT_DOUBLE_EQ(double(inst["LineExposureDuration"]), 50.88);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2009-06-30T08:34:35.424411" );
+  ASSERT_EQ(inst["StopTime"][0].toStdString(), "2009-06-30T08:34:35.678811" );
+  ASSERT_EQ(inst["SpacecraftYawDirection"][0].toStdString(), "FORWARD" );
+  ASSERT_EQ(inst["OrbitLimbDirection"][0].toStdString(), "ASCENDING" );
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  ASSERT_EQ(archive["ProductId"][0].toStdString(), "M3T20090630T083407_V03_RDN" );
+  ASSERT_EQ(archive["SourceProductId"][0].toStdString(), "M3T20090630T083407_V01_L0.IMG" );
+  ASSERT_EQ(archive["ProductType"][0].toStdString(), "CALIBRATED_IMAGE" );
+
+  // BandBin Group
+  // Check size, first, 2 middle, and last values? Enough?
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  ASSERT_EQ(bandbin["Center"].size(), 256);
+  ASSERT_EQ(bandbin["Width"].size(), 256);
+  ASSERT_EQ(bandbin["FilterNumber"].size(), 256);
+  ASSERT_EQ(bandbin["OriginalBand"].size(), 256);
+
+  ASSERT_DOUBLE_EQ(bandbin["Center"][0].toDouble(), 446.02);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][64].toDouble(), 1084.8);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][128].toDouble(), 1723.5899999999999);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][255].toDouble(), 2991.17);
+
+  ASSERT_DOUBLE_EQ(bandbin["Width"][0].toDouble(), 12.31);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][64].toDouble(), 12.29);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][128].toDouble(), 12.61);
+  ASSERT_DOUBLE_EQ(bandbin["Width"][255].toDouble(), 12.18);
+
+
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][0].toDouble(), 5);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][64].toDouble(), 69);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][128].toDouble(), 133);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][255].toDouble(), 260);
+
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][0].toDouble(), 1);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][64].toDouble(), 65);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][128].toDouble(), 129);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][255].toDouble(), 256);
+
+  // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -86520);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_DOUBLE_EQ(hist->Average(), 15.58169779027077);
+  ASSERT_DOUBLE_EQ(hist->Sum(), 47368.361282423139);
+  ASSERT_EQ(hist->ValidPixels(), 3040);
+  ASSERT_DOUBLE_EQ(hist->StandardDeviation(), 2.2696592481066249);
+
+  delete hist;
+}
+
+
+TEST(Chan1m32Isis, Chan1m32IsisTestFowardDescending) {
```

Comment:
I'm not sure if we need to test all of the same keywords + DNs for each of the forward/reverse ascending/descending, but I wrote the tests anyway.  We can trim this down as needed.